libyul/YulStack.cpp

```diff
@@ -428,3 +441,15 @@ void YulStack::reportUnimplementedFeatureError(UnimplementedFeatureError const&
 	solAssert(_error.comment(), "Unimplemented feature errors must include a message for the user");
 	m_errorReporter.unimplementedFeatureError(1920_error, _error.sourceLocation(), *_error.comment());
 }
+
+void YulStack::reportCodeGenerationError(Error const& _error)
+{
+	// CodeGenerationErrors can be reported after analysis.
+	// No error reporter available then so they're thrown instead.
+	solAssert(_error.type() == Error::Type::CodeGenerationError);
+	solAssert(!_error.secondarySourceLocation());
+	if (!_error.sourceLocation())
+		m_errorReporter.codeGenerationError(_error.errorId(), _error.what());
+	else
+		m_errorReporter.codeGenerationError(_error.errorId(), *_error.sourceLocation(), _error.what());
+}
```

Comment:
Can't this be a public method of [ErrorReporter](https://github.com/ethereum/solidity/blob/df91531c3aecf6477613874113a783bfb99ec6d0/liblangutil/ErrorReporter.h#L39)? So you could eliminate the duplication here and in CompilerStack.

Reply:
Maybe it could, but for some reason `ErrorReporter` tries hard not to expose full `Error` object in its interface I did not want to get side-tracked by figuring out why. Maybe there's a good reason for that.

Also, these methods will not necessarily be the same between `CompilerStack` and `YulStack`. The source locations will probably need to be handled differently. Maybe I will need to clear them when passing errors from `YulStack` to `CompilerStack`.

Reply:
I did that now and it eliminated the duplicated functions, but I'll have to duplicate things again soon to properly deal with locations - this has to work differently in `CompilerStack` and `YulStack`. So this is only a temporary win.